flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JdbcBatchingOutputFormat.java

```diff
@@ -105,6 +119,66 @@ public void open(int taskNumber, int numTasks) throws IOException {
 		}
 	}
 
+	private void tryCreateTableIfNotExists(int taskNumber, JdbcDdlOptions jdbcDdlOptions) throws SQLException, ClassNotFoundException {
+		// only try to create table from one TM, simply choice the minimum TM as leader
+		// after FLIP-27，this can be refactor better.
+		final boolean isCreateTableLeader = taskNumber == 0 ? true : false;
+		final JDBCOptions jdbcOptions = jdbcDdlOptions.getJdbcOptions();
+		final int maxRetries = jdbcDdlOptions.getMaxRetry();
+
+		int retry = 1;
+		if (isCreateTableLeader) {
+			while (retry <= maxRetries) {
+				try (Connection conn = new SimpleJdbcConnectionProvider(jdbcOptions).getConnection(); Statement st = conn.createStatement()) {
+					st.execute(jdbcDdlOptions.getCreateTableStatement());
+					break;
+				} catch (ClassNotFoundException e) {
+					throw new RuntimeException(
+						String.format("JDBC Driver class %s can not found.", jdbcOptions.getDriverName()), e);
+				} catch (SQLException e) {
+					if (retry == maxRetries) {
+						throw new RuntimeException(
+							String.format("Create table %s failed after %s attempts.", jdbcOptions.getTableName(), retry), e);
+					}
+					try {
+						Thread.sleep(1000 * retry);
+					} catch (InterruptedException ine) {
+						Thread.currentThread().interrupt();
+						throw new RuntimeException("Interrupted while retry.", ine);
+					}
+				}
+			}
+			retry++;
+		} else {
+			// just wait leader to create table
+			while (retry <= maxRetries) {
+				try {
+					Thread.sleep(1000 * retry);
+				} catch (InterruptedException ine) {
+					Thread.currentThread().interrupt();
+					throw new RuntimeException("Interrupted while retry.", ine);
+				}
+				try (Connection conn = new SimpleJdbcConnectionProvider(jdbcOptions).getConnection();) {
+					DatabaseMetaData dbMeta = conn.getMetaData();
+					ResultSet tables = dbMeta.getTables(null, null, jdbcOptions.getTableName(), null);
+					if (tables.next()) {
+						return;
```

Comment:
We can extract this method. The task who creates table also needs this method to check table existence first.